### PR TITLE
add waveshare_esp32_touch_amoled_241

### PR DESCRIPTION
**Pull request by wurongmin**
Component name: Waveshare_ESP32_TOUCH_AMOLED_241, for use in md files. I would like the board name to be Waveshare ESP32-Touch-AMOLED-2.41.
This application will provide us with great publicity. The board has not been officially launched yet, but I can add the link once it is live. Our company's official website: https://www.waveshare.net/.

### CODE REVIEW

**Comment by wurongmin** [comment]
Why is there a red error? Does it indicate that manual review is needed?

**Comment by me-no-dev** [comment]
missing proper declaration for UART and SPI pins

**Comment by me-no-dev** [comment]
Also: https://github.com/espressif/arduino-esp32/actions/runs/10847091674/job/30102292599?pr=10342#step:3:16

File: boards.txt
Comment (P-R-O-C-H-Y) [inline boards.txt:1]: Also there is a problem that "." is not a valid character for board naming.
`waveshare_esp32_touch_amoled_2.41` needs to be changed to something like `waveshare_esp32_touch_amoled_2_41`

**Comment by wurongmin** [comment]
I modified the UART, SPI, and the name suffix.

File: boards.txt
Comment (wurongmin) [reply boards.txt:1]: OK,Thank you

Comment (wurongmin) [reply boards.txt:1]: I modified the UART, SPI, and the name suffix.

**Review by P-R-O-C-H-Y — APPROVED** [review]
LGTM

**Pull request merged**